game/view/sector.lua

```diff
@@ -223,11 +224,92 @@ function SectorView:setCursorPos(i,j)
   self.cursor.j = j
 end
 
+--Function checks if a target position is inside "target cone" given desired direction {di,dj}
+local function _isInCone(origin_i, origin_j, target_i, target_j, di, dj)
+  local delta_i = target_i - origin_i
+  local delta_j = target_j - origin_j
+
+  if     di == -1 and dj == 0 then   --UP
+    return (delta_j < -2*delta_i and delta_j > 2*delta_i)
+  elseif di == 0 and dj == 1 then   --RIGHT
+    return (delta_i < 2*delta_j and delta_i > -2*delta_j)
+  elseif di == 1 and dj == 0 then   --DOWN
+    return (delta_j < 2*delta_i and delta_j > -2*delta_i)
+  elseif di == 0 and dj == -1 then     --LEFT
+    return (delta_i < -2*delta_j and delta_i > 2*delta_j)
+  elseif di == -1 and dj == 1 then   --UPRIGHT
+    return (delta_i >= -2*delta_j and 2*delta_i <= -delta_j)
```

Comment:
I think the diagonals's inequations are off, but as you said, it works somehow. I would suggest fixing them just so we can try to see how it feels, though.